### ISSUE TITLE
linter: jest/no-conditional-expect false positive for async expects

### ISSUE THREAD

**Issue by GerroDen** [oxc-project/oxc#9026]
### What version of Oxlint are you using?

0.15.10

### What command did you run?

oxlint -c ../../packages/linting/oxlintrc.json

### What does your `.oxlintrc.json` config file look like?

```jsonc
{
  "$schema": "../../node_modules/oxlint/configuration_schema.json",
  "plugins": ["import", "jest", "react-perf", "react", "unicorn", "typescript", "oxc"],
  "rules": {
    "jest/valid-title": ["error", { "ignoreTypeOfDescribeName": true, "ignoreTypeOfTestName": true }]
  }
}
```

### What happened?

The test looks as follows
```
    it("throws error", async () => {
     const fn = async () => throw new Error();

      const promise = fn();

      await expect(promise).rejects.toThrow();
    });
```
and the linter reports
```
  ⚠ eslint-plugin-jest(no-conditional-expect): Unexpected conditional expect
    ╭─[test.spec.ts:29:13]
 28 │ 
 29 │       await expect(promise).rejects.toThrow(MissingIterationsError);
    ·             ──────
 30 │     });
    ╰────
```
But this totally fine jest / vitest code to test promises.

The documentation states this is incorrect use
> it("throws an error", async () => {
  await foo().catch((error) => expect(error).toBeInstanceOf(error));
});

But it is correct use and is not included in the original docs of the rule: https://github.com/jest-community/eslint-plugin-jest/blob/main/docs/rules/no-conditional-expect.md
